keel-core/src/main/kotlin/com/netflix/spinnaker/keel/jackson/SerializationExtensionRegistry.kt

```diff
@@ -0,0 +1,40 @@
+package com.netflix.spinnaker.keel.jackson
+
+import com.fasterxml.jackson.databind.JsonDeserializer
+import com.fasterxml.jackson.databind.JsonSerializer
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+/**
+ * A registry to allow customization of JSON serializers and deserializers .
+ */
+@Component
+class SerializationExtensionRegistry {
```

Comment:
@robfletcher I debated with myself whether to just add these new methods to `ExtensionRegistry` or create a separate class. In the end I chose a separate class trying to stick with the single-responsibility principle, but it may be overkill. If nothing else, the name is kinda long and it's another thing to wire in, in various places. Thoughts?

Reply:
I personally like it when config stuff is all together in one file or package because sometimes you're searching for the config that you vaguely know exists but you can't for the life of you remember what it is called. 

That's not a reason to change it here, I don't mind either way. But since people rarely remember that there are settings controlling things 6 months down the line it's just a thought.

Reply:
I'm struggling to see that this class provides much value at all. Isn't it just adding indirection to the calls the `*Module` classes make? Where modules were previously calling the Jackson API directly (which seems reasonable since modules are a Jackson concept) now they're calling this and then you have to loop over all the things that were registered later to make the same calls to the Jackson API. It doesn't look like this is queried anywhere else. 

The purpose of the original `ExtensionRegistry` was that we needed to have a way to a) register subtypes with Jackson, and b) make the schema generator aware of those same subtypes. The `ExtensionRegistry` allows us to do both those things with one call so you don't accidentally do one and not the other. I'm not sure this class provides any similar value.

I guess what I'm saying is why is having this better than just calling the Jackson API directly?

Reply:
My thinking (which connects with Emily's comment above) was that I wanted the registration of serializers/deserializers to happen in a single place, specifically in that `registerKeelEc2ApiModule` function, but in order to do that I needed some place for the components in `keel-nflx` to register our own (de)serializers. The other challenge was that I wanted `SecurityGroupRuleDeserializer` to be registered by default in the OSS implementation, but have a way for `keel-nflx` to register a different deserializer first to replace it.

I guess I could wire in the list of `ObjectMapper` in the internal implementation and look for that particular one and replace it with ours, if that's available in the Jackson API? I dunno, I wanted to create something that wasn't serving just this one single use case, but maybe it's overkill.

Reply:
Would just having a module in keel-nflx address that, or would something still be missing?